src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,68 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final DateTimeFormatter DATE_FORMATTER =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data,
+                                String dateFrom, String dateTo) {
+        LocalDate from;
+        LocalDate to;
+
+        try {
+            from = LocalDate.parse(dateFrom, DATE_FORMATTER);
+            to = LocalDate.parse(dateTo, DATE_FORMATTER);
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("Invalid date format: "
+                    + dateFrom + " or " + dateTo, e);
+        }
+
+        int[] salaries = new int[names.length];
+
+        for (String record : data) {
+            String[] parts = record.split(" ");
+            if (parts.length < 4) {
+                continue;
+            }
+
+            try {
+                LocalDate currentDate = LocalDate.parse(parts[0], DATE_FORMATTER);
+                if (!currentDate.isBefore(from) && !currentDate.isAfter(to)) {
+                    String employeeName = parts[1];
+                    for (int i = 0; i < names.length; i++) {
+                        if (names[i].equals(employeeName)) {
+                            int hours = Integer.parseInt(parts[2]);
+                            int rate = Integer.parseInt(parts[3]);
+                            salaries[i] += hours * rate;
+                            break;
+                        }
+                    }
+                }
+            } catch (DateTimeParseException | NumberFormatException e) {
```

Comment:
Do you really should skip raw if it has an invalid parameter?  better throw exception